pkg/controller/core/workload_controller.go

```diff
@@ -180,22 +186,18 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	} else {
 		var updated, evicted bool
 		message := "The workload is deactivated"
+		dtCond := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadDeactivationTarget)
 		if !apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadEvicted) {
-			// The deactivation reason could be deduced as the maximum number of re-queuing retries if the workload met all criteria below:
-			//  1. The waitForPodsReady feature is enabled, which means that it has "PodsReady" condition.
-			//  2. The workload has already exceeded the PodsReadyTimeout.
-			//  3. The workload already has been re-queued previously, which means it doesn't have the requeueAt field.
-			//  4. The number of re-queued has already reached the waitForPodsReady.requeuingBackoffLimitCount.
-			if apimeta.IsStatusConditionFalse(wl.Status.Conditions, kueue.WorkloadPodsReady) &&
-				((wl.Status.RequeueState == nil && ptr.Equal(r.waitForPodsReady.requeuingBackoffLimitCount, ptr.To[int32](0))) ||
-					(wl.Status.RequeueState != nil && wl.Status.RequeueState.RequeueAt == nil &&
-						ptr.Equal(wl.Status.RequeueState.Count, r.waitForPodsReady.requeuingBackoffLimitCount))) {
+			if dtCond != nil && dtCond.Reason == kueue.WorkloadRequeuingBackoffLimitExceeded {
 				message = fmt.Sprintf("%s due to exceeding the maximum number of re-queuing retries", message)
```

Comment:
Can we take the message from the condition, and build when adding the condition?

Reply:
Done